lib/nmatrix/jruby/math.rb

```diff
@@ -1067,6 +1067,33 @@ def scale(alpha, incx=1, n=nil)
   alias :permute_columns  :laswp
   alias :permute_columns! :laswp!
 
+  #
+  # call-seq:
+  #     positive_definite? -> boolean
+  #
+  # A matrix is positive definite if it’s symmetric and all its eigenvalues are positive
+  #
+  # * *Returns* :
+  #   - A boolean value telling if the NMatrix is positive definite or not.
+  # * *Raises* :
+  #   - +ShapeError+ -> Must be used on square matrices.
+  #
+
+  def positive_definite?
+    raise(ShapeError, "positive definite calculated only for square matrices") unless
+      self.dim == 2 && self.shape[0] == self.shape[1]
+    ans = true
+    cond = 0
+    while cond != self.cols
+      if self[0..cond, 0..cond].det <= 0
+        ans = false
+        break
```

Comment:
Instead of doing `ans = false` and `break`, combine these two lines into `return false`. Then you can eliminate the `ans` variable altogether and just make the last line of the method `true`.